### PR TITLE
Allowing a task to pass a _new_state

### PR DESCRIPTION
**Pull request by drtriumph**
Allowing a task to pass a `_new_state` in the object passed to `resolve` that overrides the task spec's `finished_state`

### CODE REVIEW

File: src/lib/queue_worker.js [hunk 1]
```diff
@@ -174,15 +174,25 @@ QueueWorker.prototype._resolve = function(taskNumber) {
...
+          if (!_.isPlainObject(outputTask)) {
+            outputTask = {};
           }
-          newTask._state = self.finishedState;
-          newTask._state_changed = Firebase.ServerValue.TIMESTAMP;
-          newTask._owner = null;
-          newTask._progress = 100;
-          newTask._error_details = null;
-          return newTask;
+          if (outputTask._new_state === false) {
```
Comment (mbleigh) [inline src/lib/queue_worker.js]: Let's put a comment here explaining what exactly this does.

Comment (mbleigh) [inline src/lib/queue_worker.js]: Also let's move this conditional up with the `self.finishedState` null check.

File: src/lib/queue_worker.js [hunk 2]
```diff
@@ -171,18 +171,24 @@ QueueWorker.prototype._resolve = function(taskNumber) {
...
+          var outputTask = _.clone(newTask);
+          if (!_.isPlainObject(outputTask)) {
+            outputTask = {};
           }
-          if (!_.isPlainObject(newTask)) {
-            newTask = {};
+          outputTask._state = _.get(outputTask, '_new_state');
+          delete outputTask._new_state;
+          if (!_.isNull(outputTask._state) && !_.isString(outputTask._state)) {
+            if (_.isNull(self.finishedState) || outputTask._state === false) {
```
Comment (mbleigh) [inline src/lib/queue_worker.js]: Can you add a comment about the explicit `false` API just because it's not necessarily intuitive.

Comment (drtriumph) [inline src/lib/queue_worker.js]: :+1:

**Comment by mbleigh** [comment]
LGTM.

Let the Shipping Begin

**Pull request merged**